pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -503,11 +516,85 @@ if [ "$action" = dry-build ]; then
     extraBuildFlags+=(--dry-run)
 fi
 
+if [ "$action" = list-generations ]; then
+    if [ ! -L "$profile" ]; then
+        log "No profile \`$(basename "$profile")' found" >&2
+        exit 1
+    fi
+
+    generation_from_dir() {
+        generation_dir="$1"
+        generation_base="$(basename "$generation_dir")" # Has the format "system-123-link" for generation 123
+        no_link_gen="${generation_base%-link}"  # remove the "-link"
+        echo "${no_link_gen##*-}" # remove everything before the last dash
+    }
+    describe_generation(){
+        generation_dir="$1"
+        generation_number="$(generation_from_dir "$generation_dir")"
+        nixos_version="$(cat "$generation_dir/nixos-version" 2> /dev/null || echo "Unknown")"
```

Comment:
I think bash can directly read the file without catting.

Reply:
Good point! I didn't think of that!
Though I have troubles ignoring the error with the redirect (I tried `a="$(<does_not_exist.txt 2> /dev/null)"` and it still prints "No such file or directory"). Is that even something to consider with the `nixos-version` file? I might have been a bit too defensive in the first place, but I don't know...